lib/sdm/maven.ts

```diff
@@ -0,0 +1,163 @@
+    name: "MavenGenerator",
+    intent: "create maven project",
+    description: "Creates a new Maven project",
+    tags: ["maven"],
+    autoSubmit: true,
+    parameters: SpringProjectCreationParameterDefinitions,
+    startingPoint: GitHubRepoRef.from({ owner: "atomist-seeds", repo: "spring-rest", branch: "master" }),
+    transform: [
+        UpdateReadmeTitle,
+        ...TransformMavenSpringBootSeedToCustomProject,
+    ],
+};
+// atomist:code-snippet:end
+
+export const configuration = configure(async sdm => {
+
+    // Register the generator and stop command with the SDM
+    sdm.addGeneratorCommand(MavenGenerator);
+
+    // Version goal calculates a timestamped version for the build goal
+    const versionGoal = new Version()
+        .withVersioner(MavenProjectVersioner);
+
+    // Build goal that runs "maven package", after running "mvn version" which
+    // sets a unique version for the build
+    const buildGoal = new Build(
+        { displayName: "maven build" })
+        .with({
+            name: "maven-build",
+            builder: mavenBuilder(),
+        }).withProjectListener(MvnVersion);
+
+    const mavenSpringBootRun = goal(
+        { displayName: "maven spring boot run" },
+        async gi => {
+            const { goalEvent, progressLog } = gi;
+            const port = await scanFreePort(8000, 8100);
+            const appUrl = `http://localhost:${port}`;
+
+            try {
+                const result = await execPromise(
+                    "mvn",
+                    ["spring-boot:run", `-Dspring-boot.run.arguments=--server.port=${port}`],
+                );
+                await gi.addressChannels(
+                    slackSuccessMessage(
+                        "Maven Spring Boot Run",
+                        `Successfully started ${codeLine(goalEvent.sha.slice(0, 7))} at ${url(appUrl)}`,
+                        {},
+                    ),
+                    { });
+
+                return {
+                    state: SdmGoalState.success,
+                    externalUrls: [
+                        { label: "http", url: appUrl },
+                    ],
+                };
+
+            } catch (e) {
+                progressLog.write(`Container run command failed: %s`, e.message);
+                return {
+                    code: 1,
+                };
+            }
+        },
+    );
+
+    // This SDM has three PushRules: no goals, build and docker
+    return {
+        no_goals: {
+            test: not(hasFile("pom.xml")),
+            goals: DoNotSetAnyGoals.andLock(),
+        },
+        build: {
+            goals: [
+                buildGoal,
+            ],
+        },
+        run: {
+            dependsOn: "build",
+            goals: [
+                mavenSpringBootRun,
+            ],
+        },
+    };
+}, { name: "maven" });
+
+/**
+ * Read the Docker hostname from the DOCKER_HOST environment variable
+ */
+function readDockerHost(): string | undefined {
```

Comment:
I don't think we need this method any longer?

Reply:
Removed